lib/consolidate.js

```diff
@@ -91,13 +93,24 @@ function read(path, options, fn) {
   if (cached) return fn(null, str);
 
   // read
-  fs.readFile(path, 'utf8', function(err, str){
-    if (err) return fn(err);
-    // remove extraneous utf8 BOM marker
-    str = str.replace(/^\uFEFF/, '');
-    if (options.cache) readCache[path] = str;
-    fn(null, str);
-  });
+  if (nowReading[path]) {
+    if (!readListeners[path]) readListeners[path] = [fn];
+    else readListeners[path].push(fn);
+  }else{
+    nowReading[path] = true;
+    fs.readFile(path, 'utf8', function(err, str){
+      if (err) return fn(err);
```

Comment:
I ran into this situation using the metalsmith-templates package.  I have a high number of templates and a lot of content files that use the same ones over and over.

I believe what is happening is that metalsmith-templates is asking consolidate to open the same template many times before the first file is opened and therefore cached.  The fix here queues up all the template requests for the same file instead of spawning many fs.readFile requests and then answers them after completed.

Metalsmith recently added a concurrency setting to control how many open file handles are used during a particular run.  This of course is not built into node so it doesn't extend to consolidate.  A work around is to bump up the open file limit. You may be able to reproduce this more easily by lowering it.